### ISSUE TITLE
Question comment deletion and answer commenting routes need fixes

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#2940]
Hi, @Souravirus just found a few more here:

First, deleting a comment on a question showed a 404:

`GET https://publiclab.org/comment/delete/19930?type=question&_=1530203054126 404 (Not Found)`

And updating a comment to a question (not an answer) led to a Page does not exist (404) with this URL: `https://publiclab.org/comment/update/19918?type=question&body=`

Third, I saw an error notice when posting a comment on an answer, but it was a 500 error:

```
application-4512e1778aedf64866c59f9ee9f2aa74bfae4397ff02660177746bc96b1a5f5b.js:10255 GET https://publiclab.org/comment/answer_create/1061?type=question&body=Apologies+for+the+noisy+comments+above%2C+we%27ve+found+an+error+and+are+trying+to+fix+it!+%40tester&_=1530203250156 500 (Internal Server Error)
```

I'll look for the log on that one...

@Souravirus Thanks for being so great at following up with these. I imagine we may still find a few more, but your routing tests and your attention to detail is making a big difference and we deeply appreciate it! 🙌

**Comment by jywarren** [comment]
And one more! 

`/answers/accept/931?_=1530140141188:1 Failed to load resource: the server responded with a status of 404 (Not Found)`

**Comment by jywarren** [comment]
The 500 error on posting a comment on an answer was: 

```
[741de7dd-a82b-40ec-a09e-017d187f1450] Processing by CommentController#answer_create as JS
[741de7dd-a82b-40ec-a09e-017d187f1450]   Parameters: {"type"=>"question", "body"=>"Apologies for the noisy comments above, we've found an error and are trying to fix it! 
@tester", "_"=>"1530203250156", "aid"=>"1061"}
[741de7dd-a82b-40ec-a09e-017d187f1450]   Rendering comment_mailer/notify_answer_author.html.erb
[741de7dd-a82b-40ec-a09e-017d187f1450]   Rendered comment_mailer/notify_answer_author.html.erb (2.7ms)
[741de7dd-a82b-40ec-a09e-017d187f1450] Sent mail to cfastie@gmail.com (10.6ms)
[741de7dd-a82b-40ec-a09e-017d187f1450] Completed 500 Internal Server Error in 59ms (ActiveRecord: 18.7ms)
[741de7dd-a82b-40ec-a09e-017d187f1450]   
[741de7dd-a82b-40ec-a09e-017d187f1450] NoMethodError (undefined method `id' for nil:NilClass):
[741de7dd-a82b-40ec-a09e-017d187f1450]   
[741de7dd-a82b-40ec-a09e-017d187f1450] app/mailers/comment_mailer.rb:34:in `notify_callout'
```

**Comment by jywarren** [comment]
Note that the comment-answer was saved, but an error was shown.

**Comment by jywarren** [comment]
And one more! Promoting a comment to an answer:

`/comment/make_answer/19246 Failed to load resource: the server responded with a status of 404 (Not Found)`

Sorry there are so many all at once! I just did a careful search. Your new routing tests are AWESOME and will prevent us missing these in the future!

**Comment by Souravirus** [comment]
Yeah seeing these one by one @jywarren

**Comment by Souravirus** [comment]
Hi @jywarren, the deleting of comment on a question was working fine when I tested. The comment was not deleted at that moment (I don't know if it was supposed to be deleted at that time) but when I reloaded it the comment was deleted.

**Comment by jywarren** [comment]
hmm, is there a notificatoin maybe via the new AJAX notifications? And i
think it ought to disappear visibly, even if that means we need to refresh
the page when we perform the deletion.

**Comment by Souravirus** [comment]
No there was no AJAX notification

**Comment by Souravirus** [comment]
yeah I found a routing error on promoting a comment to answer. Sending a PR soon. Also, can you tell me the steps to promote a comment to a question, so that I can test it. Thanks!!

**Comment by Souravirus** [comment]
Yeah got the way to test the promoting the comment to questions. Testing it

**Comment by Souravirus** [comment]
Also, when I tested promoting comment to a question locally it worked perfectly.

**Comment by Souravirus** [comment]
Yeah got a routing error in accepting of the answer. Sending a commit.

**Comment by jywarren** [comment]
I wasn't able to promote the 2nd comment here to an answer -- do you know why? I'll just reopen this until we can resolve it. Thanks!

https://publiclab.org/questions/marlokeno/09-20-2017/can-nasa-viirs-data-used-for-disaster-maps

**Comment by Souravirus** [comment]
Hi @jywarren, I tested updating comment to answer and it was working fine. Is the PR #2942 merged?

**Comment by jywarren** [comment]
OK, that now works. I still see that there's a refresh required before deletion of a comment on a question shows up. Would you mind taking a look? Maybe it just needs to either actually remove the HTML element upon success OR just refresh the page via Javascript. I could help with that? What do you think?

https://publiclab.org/questions/warren/06-26-2018/are-there-lists-of-plants-and-the-chemicals-they-can-remediate

**Comment by Souravirus** [comment]
Yeah I will work on it.

**Comment by Souravirus** [comment]
Hi @jywarren, actually the code was already there. The only thing that was not doing the trick was the bracket mismatch in the code. Now its working fine. Thank you!!
